src/PhpWord/Shared/Microsoft/PasswordEncoder.php

```diff
@@ -0,0 +1,205 @@
+<?php
+/**
+ * This file is part of PHPWord - A pure PHP library for reading and writing
+ * word processing documents.
+ *
+ * PHPWord is free software distributed under the terms of the GNU Lesser
+ * General Public License version 3 as published by the Free Software Foundation.
+ *
+ * For the full copyright and license information, please read the LICENSE
+ * file that was distributed with this source code. For the full list of
+ * contributors, visit https://github.com/PHPOffice/PHPWord/contributors.
+ *
+ * @see         https://github.com/PHPOffice/PHPWord
+ * @copyright   2010-2017 PHPWord contributors
+ * @license     http://www.gnu.org/licenses/lgpl.txt LGPL version 3
+ */
+
+namespace PhpOffice\PhpWord\Shared\Microsoft;
+
+/**
+ * Password encoder for microsoft office applications
+ */
+class PasswordEncoder
```

Comment:
@maartenba @Progi1984 Would it make sense to move this class to the PHPOffice common project? I guess it could be reused in PhpSpreadsheet

Reply:
I think it would make sense to do, yes.

Reply:
The same as @maartenba 

Poke @MarkBaker & @PowerKiki

Reply:
@Progi1984 @maartenba @MarkBaker @PowerKiKi I created a PR in Common for this.

Reply:
I am actually not quite sure how this should/could be used in PhpSpreadsheet...

FYI, the PR: https://github.com/PHPOffice/Common/pull/15

Reply:
It seems indeed that pre-2013 excel sheets use a very simplistic password hashing which is completely different.
In more recent specs, it also uses something very similar to what is used in word. Not sure the algorithm is the same though.

https://www.thespreadsheetguru.com/blog/2014/8/20/understanding-excels-password-security-methodology

Reply:
@PowerKiKi I see you have the pre-2013 hashing implemented already in PhpSpreadsheet/Shared/PasswordHasher, this would be a more modern password hashing mechanism